### ISSUE TITLE
R.hasPath returns false for the empty array as a path

### ISSUE THREAD

**Issue by zouxuoz** [ramda/ramda#2839]
## Bug Report

R.hasPath returns false for the empty array as a path

## To Reproduce

```
const R = require('ramda');

console.log(R.hasPath([], { value: 10 })); // false
console.log(R.path([], { value: 10 })); // { value: 10 }
```

## Expected behavior

R.hasPath should returns `true` for the empty array as a path

**Comment by CrossEye** [comment]
You're right.  This seems a problem.  Are you interested in creating a PR?

**Comment by ZerOTanK** [comment]
While trying to fix this, I noticed that hasPath isn't working for non-objects like undefined, null, etc.

You are getting `Cannot read property 'length' of null`

**Comment by sturdynut** [comment]
👋I'd like to take a stab at this.

@zouxuoz -Do you mind if I PR this?

**Comment by Bradcomp** [comment]
@sturdynut have you taken a look at the [related PR](https://github.com/ramda/ramda/pull/2841)?

There was some discussion there on the proper behavior for `hasPath`, but the implementation is complete.

**Comment by sturdynut** [comment]
Apologies, I missed that.  Thank you!
